token_dispenser/aws/launchpad_token.py

```
@@ -0,0 +1,50 @@
+"""
+This module accesses the /gettoken endpoint from launchpad application to obtain
+Launchpad Token
+"""
+import json
+from tempfile import NamedTemporaryFile
+import time
+import requests
+from token_dispenser.logging_config import shared_logger
+
+cached_cert_file: NamedTemporaryFile = None
+
+
+def get_token(url: str, cert_file: str):
+    """
+    Access the /gettoken endpoint using the provided private key and certificate to obtain a token.
+
+    :param url: The URL of the /gettoken endpoint.
+    :param cert_file: the full path to a certificate file containing both cert string and
+    private key.
+
+    :return: The token response as a dict.
+    """
+    logger = shared_logger()
+    try:
+        logger.debug(f"Trying to obtain a token from {url}")
+        # Serialize private key and certificate into PEM format
+
+        response = requests.get(
+            url,
+            cert=cert_file,
+            timeout=10  # timeout of 10 seconds
+        )
+        # Check if the request was successful
+        if response.status_code == 200:
+            logger.info("Successfully obtained token.")
+            content_str: str = response.content.decode('utf-8')
+            # Parse the string into a JSON object
+            json_data = json.loads(content_str)
+            current_time: int = int(time.time())
+            json_data['expires_at'] = current_time + int(json_data['session_maxtimeout'])
+            json_data['created_at'] = current_time
+            return json_data
+
+        logger.info("launchpad /gettoken call failed with code {response.status_code}")
+        response.raise_for_status()
+        return None
+    except Exception as exception:
+        logger.exception("get_token error occurred")
+        raise exception
```

Not sure about this generic exception handler. If we get something other than a 200, let's just throw an Exception stating that the call failed or just call `raise_for_status()` and let it boil up. Remove the handler